packages/@lwc/engine-core/src/framework/modules/static-parts.ts

```diff
@@ -84,12 +90,17 @@ export function applyStaticParts(root: Element, vnode: VStatic, renderer: Render
         return;
     }
 
-    traverseAndSetElements(root, parts, renderer); // this adds `part.elm` to each `part`
+    // This adds `part.elm` to each `part`. We have to do this on every mount/patch because the `parts`
+    // array is recreated from scratch every time, so each `part.elm` is now undefined.
+    traverseAndSetElements(root, parts, renderer);
```

Comment:
I'm not super happy with this, since it means we are traversing every time we patch the static content (which kind of defeats the purpose of the static content). But I don't have any good ideas for optimizing this right now.

Reply:
We could keep a cache of the static refs or something, but I don't know how much time this will save in practice. `traverseAndSetElements` is already pretty fast (0.3ms total for our entire Karma test suite).

Reply:
Maybe we could just build a `Map<number,Element>` once. This would be a significant refactor though – probably better for another PR.

Reply:
https://github.com/salesforce/lwc/issues/3800